File: MainApp/src/main/java/com/sleticalboy/learning/components/receiver/GlobalReceiver.java
package com.sleticalboy.learning.components.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

public class GlobalReceiver extends BroadcastReceiver {

    private static final String TAG = "GlobalReceiver";

    @Override
    public void onReceive(final Context context, final Intent intent) {
        final String action = intent.getAction();
        if (Intent.ACTION_SCREEN_ON.equals(action)
                || Intent.ACTION_USER_PRESENT.equals(action)) {
            Log.d(TAG, "onReceive() receive: " + action + ", start work.");
        }
    }
}

File: MainApp/src/main/java/com/sleticalboy/learning/components/receiver/GlobalReceiver.kt
package com.sleticalboy.learning.components.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log


class GlobalReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val action = intent.action
        if (Intent.ACTION_SCREEN_ON == action || Intent.ACTION_USER_PRESENT == action) {
            Log.d(TAG, "onReceive() receive: $action, start work.")
        }
    }

    companion object {
        private const val TAG = "GlobalReceiver"
    }
}